core/src/allocation.jl

```diff
@@ -661,6 +661,8 @@ function set_objective_priority!(
         ex = sum(problem[:F_abs])
     end
 
+    demand_max = 0
```

Comment:
```suggestion
    demand_max = 0.0
```

Later you do `demand_max = max(demand_max, d)`, which means that `demand_max` becomes either an Int or Float64.